data/sql/headers.go

```diff
@@ -120,12 +120,12 @@ const (
     WHERE height = ?
     `
 
-	sqlSelectTips = `
-	SELECT hash, height, version, merkleroot, nonce, bits, chainwork, previousblock, timestamp, cumulatedWork, header_state
+	sqlSelectHighestHundred = `
+	SELECT hash, height, version, merkleroot, nonce, bits, chainwork, previousblock, timestamp, header_state, cumulatedWork
 	FROM headers
-	WHERE hash NOT IN (SELECT previousblock
-					   FROM headers)
-				   `
+	ORDER BY height DESC
+	LIMIT 100
+	`
```

Comment:
One think still bothers me. So i went after my idea, and I think that changing only this SQL to such a query should do the work and we will still get all tips of all concurrent chains and orphans:

```suggestion
sqlSelectTips = `
with mainTip as (
select hash, height, version, merkleroot, nonce, bits, chainwork, previousblock, timestamp, header_state, cumulatedWork
from headers
where header_state = 'LONGEST_CHAIN'
order by height
limit 1
)
select hash, height, version, merkleroot, nonce, bits, chainwork, previousblock, timestamp, header_state, cumulatedWork
from mainTip
union
select hash, height, version, merkleroot, nonce, bits, chainwork, previousblock, timestamp, header_state, cumulatedWork
from headers
where header_state != 'LONGEST_CHAIN' and
        hash not in (select previousblock from headers where header_state != 'LONGEST_CHAIN')
        `
```

Reply:
For the sake of completeness, I prepared a separate PR with this proposition:
https://github.com/libsv/pulse/pull/112
Please check if this fits your needs @sirdeggen 

And sorry if this looks a little rude that I'm mingling yours PR.

Reply:
Thanks, will take a look now

Reply:
Good, thanks.